lib/spear/pool/pool_table.ex

```diff
@@ -0,0 +1,21 @@
+defmodule Spear.Pool.PoolTable do
+  use GenServer
+
+  def start_link(_opts) do
+    GenServer.start_link(__MODULE__, [])
+  end
+
+  def init([]) do
+    ets = :ets.new(__MODULE__, [:public, :named_table, {:read_concurrency, true}])
```

Comment:
Small nit but this doesn't need `public` since the table is only modified by the owning process (`protected` is the default)